File: LOO 2°Parcial/EmpleadosPolimorfismo/src/empleadospolimorfismo/TrabajadorPorHoras.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package empleadospolimorfismo;

/**
 * 
 * @author Dimitri Medina Novelo
 */
public class TrabajadorPorHoras extends Empleado{
        protected int HorasTrabajadas;
        protected double pagoXhora;
        
        public TrabajadorPorHoras()
        {
            super ();
            this.HorasTrabajadas = 0;
            this.pagoXhora = 0;
        }
        public TrabajadorPorHoras(String nombre, int cantidad, double pago) 
        {
            super(nombre);
            this.HorasTrabajadas = cantidad > 0 ? cantidad : 0;
            this.pagoXhora = pago > 0 ? pago : 0;
        }
        public int getHrsTrabajadas()
        {
            return this.HorasTrabajadas;
        }
        
        public void setHrsTrabajadas(int cantidad)
        {
            this.HorasTrabajadas = cantidad > 0 ? cantidad : 0;
        }
        
        public double getPagoxHora()
        {
            return this.pagoXhora;
        }
        
        public void setPagoxHora(double pago)
        {
            this.pagoXhora = pago > 0 ? pago : 0;
        }
        @Override
        public double pagoDiario()
        {
            return this.HorasTrabajadas * this.pagoXhora;
        }
        
        @Override
        public void imprimir()
        {
            System.out.println("Empleado tipo Trabajador por horas");
            System.out.println("NOMBRE: " + this.nombre + "\nPAGO POR HORA: " + this.pagoXhora + "\nCANTIDAD HORAS TRABAJADAS: " + this.HorasTrabajadas + "\nPAGO DIARIO: " + this.pagoDiario());
        }
}

File: LOO 2°Parcial/EmpleadoEjercicio1cpp/EmpleadoEjercicio1cpp/TrabajadorPorHoras.cpp
#include "TrabajadorPorHoras.h"

TrabajadorPorHoras::TrabajadorPorHoras() : Empleado() {
	this->HorasTrabajadas = 0;
	this->pagoXhora = 0;
}

TrabajadorPorHoras::TrabajadorPorHoras(string nombre, int cantidad, double pago) : Empleado(nombre) {
	this->HorasTrabajadas = cantidad > 0 ? cantidad : 0;
	this->pagoXhora = pago > 0 ? pago : 0;
}

int TrabajadorPorHoras::getHrsTrabajadas(void) {
	return this->HorasTrabajadas;
}

void TrabajadorPorHoras::setHrsTrabajadas(int cantidad) {
	this->HorasTrabajadas = cantidad > 0 ? cantidad : 0;
}

double TrabajadorPorHoras::getPagoxHora(void) { 
	return this->pagoXhora;
}

void TrabajadorPorHoras::setPagoxHora(double pago) { 
	this->pagoXhora = pago > 0 ? pago : 0;
}

double TrabajadorPorHoras::pagoDiario(void) { return this->HorasTrabajadas * this->pagoXhora; }

void TrabajadorPorHoras::imprimir(void) {
	cout << "Empleado tipo Trabajador por horas" << endl;
	cout << "NOMBRE: " << this->nombre << "\nPAGO POR HORA: " << this->pagoXhora << "\nCANTIDAD HORAS TRABAJADAS: " << this->HorasTrabajadas << "\nPAGO DIARIO: " << this->pagoDiario() << endl;
}